mod/node-core/pkg/components/payload_builder.go

```diff
@@ -32,32 +32,87 @@ import (
 
 // LocalBuilderInput is an input for the dep inject framework.
 type LocalBuilderInput[
+	AttributesFactoryT AttributesFactory[BeaconStateT, PayloadAttributesT],
```

Comment:
Attributes factory shouldn't be a generic type it should be an interface, since its primary purpose is to perform business logic NOT be a type